rpc-ipc.md

```diff
@@ -0,0 +1,249 @@
+
+## Server-side Tracing
+
+The goals of server-side tracing are to trace the lifetime of a request to a server and connect this instrumentation to any pre-existing trace in the client. You can do this by creating spans when the server receives the request and ending the span when the server finishes processing the request. The workflow for tracing a server request is as follows:
+
+* Server Receives Request
+    * Extract the current trace state from the inter-process transport (HTTP, etc)
+    * Start the span
+    * Store the current trace state
+* Server Finishes Processing the Request / Sends Response
+    * Finish the span
+
+Because this workflow depends on request processing, you'll need to know how to  change the  framework’s requests and responses handling--whether this is through filters, middleware, a configurable stack, or some other mechanism.
+
+### Extract the Current Trace State
+
+In order to trace across process boundaries in distributed systems, services need to be able to continue the trace injected by the client that sent each request. OpenTracing allows this to happen by providing inject and extract methods that encode a span's context into a carrier. (The specifics of the encoding is left to the implementor, so you won't have to worry about that.)
+
+If there was an active request on the client side, the span context will already be injected into the the request. Your job is to then extract that span context using the io.opentracing.Tracer.extract method. The carrier that you'll extract from depends on which type of service you're using; web servers, for example, use HTTP headers as the carrier for HTTP requests (as shown below):
+
+Python:
+
+```Python
+span_ctx = tracer.extract(opentracing.Format.HTTP_HEADERS, request.headers)
+```
+
+Java:
+
+```Java
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+
+Map<String, String> headers = request.getHeaders();
+SpanContext parentSpan = tracer.getTracer().extract(Format.Builtin.HTTP_HEADERS,
+    new TextMapExtractAdapter(headers));
+```
+
+OpenTracing can throw errors when an extract fails due to no span being present, so make sure to catch the errors that signify there was no injected span and not crash your server. This often just means that the request is coming from a third-party (or untraced) client, and the server should start a new trace.
+
+### Start the span
+
+Once you receive a request and extract any existing span context, you should immediately start a span representing the lifetime of the request to the server. If there is an extracted span context present, then the new server span should be created with a ChildOf reference to the extracted span, signifying the relationship between the client request and server response. If there was no injected span, you'll just start a new span with no references.
+
+Python:

```

Comment:
don't exactly know how to do it, but it would be nice to have a style that renders different languages as tabs, so that only one language is seen at a time, making presentation easier